### PR TITLE
minor jQuery.hasclass removal

### PR DESCRIPTION
**Pull request by jwilson8767**

### CODE REVIEW

**Comment by XhmikosR** [comment]
Are there any other instances?

**Comment by XhmikosR** [comment]
Hmm tests fail though.

**Comment by jwilson8767** [comment]
No other occurrences of this particular issue, not sure why tests are failing, could `this._dialog` sometimes be a jquery instance?

**Comment by XhmikosR** [comment]
Yeah, maybe tests need to be adapted too.

**Comment by Johann-S** [comment]
That's because a lot of modals in our tests do not have a `.modal-dialog` child.

It wasn't an issue before because jQuery fail in silence but not Vanilla JS

**Comment by XhmikosR** [comment]
@jwilson8767 can you fix the issue?

**Comment by jwilson8767** [comment]
@XhmikosR Done, and tests are now passing.

**Comment by XhmikosR** [comment]
Thanks! @Johann-S I think it's fine now, if so squash and merge please.

**Comment by ghiscoding** [comment]
Hey guys, I just happened to have a peak at the PR and, I just noticed that the code is not exactly 100% bullet proof. I am using TypeScript a lot lately and in few occasion with the TSlint `no null` option and this PR would fail it. 

Technically the following is 100% safe

```diff
- if (this._dialog && this._dialog.classList.contains(ClassName.SCROLLABLE)) {
+ if (this._dialog && this._dialog.classList && this._dialog.classList.contains && this._dialog.classList.contains(ClassName.SCROLLABLE)) {
```

But I think that if you have a `classList`, you most probably have a `contains`, so the previous might be overkill and it might be more than enough to go with the simpler
```diff
- if (this._dialog && this._dialog.classList.contains(ClassName.SCROLLABLE)) {
+ if (this._dialog && this._dialog.classList && this._dialog.classList.contains(ClassName.SCROLLABLE)) {
```

Anywho, if you think my suggestion is still overkill then please disregard, but I think it's always better to avoid any possibility of errors thrown in the console... Bootstrap is awesome, thank you :)

File: js/src/modal.js [hunk 1]
```diff
@@ -245,7 +245,7 @@ class Modal {
     this._element.removeAttribute('aria-hidden')
     this._element.setAttribute('aria-modal', true)
 
-    if ($(this._dialog).hasClass(ClassName.SCROLLABLE)) {
+    if (this._dialog && this._dialog.classList.contains(ClassName.SCROLLABLE)) {
```
Comment (Johann-S) [inline js/src/modal.js]:
```js
if (this._dialog.classList.contains(ClassName.SCROLLABLE))
```

is enough, because a Modal must have a `dialog`

**Review by Johann-S — CHANGES_REQUESTED** [review]

**Comment by Johann-S** [comment]
Hi @ghiscoding,

Defensive code isn't something we should do here, because a modal must have a dialog, and when you have an `HTMLElement` you have a `classList`.

This PR won't fail in TS, because `_dialog` is an `HTMLElement`

**Review by Johann-S — APPROVED** [review]

**Comment by Johann-S** [comment]
Thanks for your work @jwilson8767 👍

**Comment by jwilson8767** [comment]
@Johann-S Thanks for giving this your attention! Looking forward to being off of jQuery :)

**Pull request merged**